File: Design Patterns/src/memento/CareTaker.java
package memento;

import java.util.ArrayList;
import java.util.List;

public class CareTaker {

    private final List<Memento> list;

    public CareTaker() {
        this.list = new ArrayList<>();
    }

    public void save(Memento pm) {
        list.add(pm);
    }

    public void load(int index) {
        list.get(index).load();
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder("Saves:\n");
        list.forEach(pm -> sb.append(pm).append("\n"));
        return sb.toString();
    }


}

File: Design Patterns/basic-patterns-java-kotlin/memento/player/CareTaker.kt
package memento.player

import java.lang.StringBuilder
import java.util.ArrayList
import java.util.function.Consumer

class CareTaker {
    private val list: MutableList<Memento>
    fun save(pm: Memento) {
        list.add(pm)
    }

    fun load(index: Int) {
        list[index].load()
    }

    override fun toString(): String {
        val sb = StringBuilder("Saves:\n")
        list.forEach(Consumer { pm: Memento? -> sb.append(pm).append("\n") })
        return sb.toString()
    }

    init {
        list = ArrayList()
    }
}